src/Sentry/Scope.cs

```diff
@@ -245,6 +245,28 @@ public void AddBreadcrumb(Breadcrumb breadcrumb)
         /// </summary>
         public void AddAttachment(Attachment attachment) => _attachments.Add(attachment);
 
+
+        /// <summary>
+        /// Clear all Attachments.
+        /// </summary>
+        public void ClearAttachments()
+        {
+#if NETCOREAPP3_0_OR_GREATER || NETSTANDARD2_1 || NET5_0_OR_GREATER
+            _attachments.Clear();
+#else
+            var oldAttachments = Interlocked.Exchange(ref _attachments, new());
+
+            while (!oldAttachments.IsEmpty)
+            {
+                if (!oldAttachments.TryTake(out _))
+                {
+                    Options.DiagnosticLogger?.LogWarning("Failed to complete the Attachments cleanup.");
+                    break;
+                }
+            }
```

Comment:
Why do we need this if we already re-assigned the field?

Reply:
It's done that way so both NET 5 and this version behave the same way.
I noticed that if you simply do 
`_ = Interlocked.Exchange(ref _attachments, new());`
It'll rely on GC to release the old attachments 
![image](https://user-images.githubusercontent.com/8229322/124338516-ce64b200-db7e-11eb-998d-a83d132654bc.png)
While TryTake will release the objects from memory after completing ClearAttachments (Following the same pattern of _attachments.Clear();)
![image](https://user-images.githubusercontent.com/8229322/124338612-53e86200-db7f-11eb-8423-ac8ba0564b6b.png)

Reply:
Thinking about that, we don't actually need that log, in any case, the data will be gone there or when the Garbage collector decides to release it.

Reply:
Yeah, I'd probably just let the GC take care of it and not stress about it too much. As long as the reference to the previous instance of `_attachments` is gone, it will get rid of the items eventually.